File: scala/integration/packagesearch/src/org/jetbrains/plugins/scala/packagesearch/PackageSearchSbtBundle.scala
package org.jetbrains.plugins.scala.packagesearch

import com.intellij.DynamicBundle
import com.jetbrains.packagesearch.intellij.plugin.extensibility.BuildSystemType
import org.jetbrains.annotations.{Nls, PropertyKey}
import org.jetbrains.plugins.scala.NlsString

import scala.annotation.varargs

object PackageSearchSbtBundle {
  private final val BUNDLE = "messages.PackageSearchSbtBundle"
  private object INSTANCE extends DynamicBundle(BUNDLE)

  val buildSystemType = new BuildSystemType(
    "SBT",
    "scala",
    "sbt-scala"
  )

  //noinspection ReferencePassedToNls
  @Nls
  @varargs
  def message(@PropertyKey(resourceBundle = BUNDLE) key: String, params: Any*): String =
    INSTANCE.getMessage(key, params.map(_.toString): _*)

  //noinspection ReferencePassedToNls
  @Nls
  @varargs
  def nls(@PropertyKey(resourceBundle = BUNDLE) key: String, params: Any*): NlsString =
    NlsString(INSTANCE.getMessage(key, params.map(_.toString): _*))
}

File: scala/integration/packagesearch/src/org/jetbrains/plugins/scala/packagesearch/PackageSearchSbtBundle.java
package org.jetbrains.plugins.scala.packagesearch;

import com.intellij.DynamicBundle;
import org.jetbrains.annotations.Nls;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.PropertyKey;

public final class PackageSearchSbtBundle {
    @NonNls
    private static final String BUNDLE = "messages.PackageSearchSbtBundle";
    private static final DynamicBundle INSTANCE = new DynamicBundle(PackageSearchSbtBundle.class, BUNDLE);

    private PackageSearchSbtBundle() {
    }

    @Nls
    public static String message(@NotNull @PropertyKey(resourceBundle = BUNDLE) String key, @NotNull Object... params) {
        return INSTANCE.getMessage(key, params);
    }
}
